modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/ScaledFloatFieldMapper.java

```diff
@@ -749,6 +749,18 @@ protected void writeValue(XContentBuilder b, long value) throws IOException {
      */
     static double decodeForSyntheticSource(long scaledValue, double scalingFactor) {
         double v = scaledValue / scalingFactor;
+
+        // If original double value is close to infinity
+        // and rounding is performed in the direction of the same infinity
+        // it is possible to "overshoot" infinity during reconstruction.
+        // E.g. for a value close to Double.MAX_VALUE "true" scaled value is 10.5
+        // and with rounding it becomes 11.
+        // Now, because of that rounding difference, 11 divided by scaling factor goes into infinity.
+        // There is nothing we can do about it so we'll settle with infinity.
+        if (Double.isInfinite(v)) {
+            return v;
```

Comment:
Actually i think we pretty much have to do it since `Infinity` is not a valid input for scaled float. So essentially this source can not be parsed into the same(-ish) document. Happy to discuss.

Reply:
But `Double.isInfinite(Double.MAX_VALUE)` is false, iiuc? This is getting confusing..

Maybe add a unittest for this case - I assume we don't have one and only get there with random values. The test can help us reason about expected behavior.

Reply:
Yes, it is finite. We would basically "round" the infinity to closest finite value.